bin/build

```diff
@@ -42,11 +42,21 @@ else
     cp -r "${MW_LAYER}/node_modules/." "${NODE_MODULES_DIR}"
 fi
 
+# Setup user function as a node module dependency in the middleware and allow
+# the user function to be required by name.
+SF_FUNCTION_PACKAGE_NAME=$(jq -r '.name' < /workspace/package.json)
+if [[ -z "${SF_FUNCTION_PACKAGE_NAME}" ]]; then
+  echo "Salesforce function package name not defined. Does the package.json have a 'name'?"
+  exit 1
+fi
+echo -n "$SF_FUNCTION_PACKAGE_NAME" > "$MW_LAYER/env/SF_FUNCTION_PACKAGE_NAME.override"
+ln -s /workspace "$MW_LAYER/node_modules/$SF_FUNCTION_PACKAGE_NAME"
```

Comment:
is there a chance for collision here? Could they name it `typescript` and cause weird things?

Reply:
Oof, yeah. There is potential. The default module name gets defaulted to `#{user_scaffold_name}-function`, so it's not super likely but definitely possible. I can check and fail if the package name is already installed.